### PR TITLE
Add support for rfc 2833

### PR DESCRIPTION
**Pull request by chukitow**
Currently SIPJS supports RFC 2833 for DTMF (https://github.com/onsip/SIP.js/pull/511)
I think will be nice that  JSIP supports it too 😄 

Based on the comments that I found https://github.com/versatica/JsSIP/pull/496 the `sendDtmf`
could take an extra option to define the DTMF type instead of  UA level.

### CODE REVIEW

File: lib/RTCSession.js [hunk 1]
```diff
@@ -1042,13 +1045,21 @@ module.exports = class RTCSession extends EventEmitter
...
-
-        options.eventHandlers = {
-          onFailed : () => { this._tones = null; }
-        };
-        dtmf.send(tone, options);
-        timeout = duration + interToneGap;
+        switch (type)
+        {
+          case RFC2833:
+            this._getDTMFRTPSender().insertDTMF(tone, duration, interToneGap);
```
Comment (Krivega) [inline lib/RTCSession.js]: if the method returns an undefined? `this._getDTMFRTPSender()`

File: lib/RTCSession.js [hunk 2]
```diff
@@ -3523,4 +3534,21 @@ module.exports = class RTCSession extends EventEmitter
...
     });
   }
+
+  _getDTMFRTPSender()
+  {
+    const streams = this._connection.getLocalStreams();
+
+    if (streams.length > 0)
+    {
+      const audioTracks = streams[0].getAudioTracks();
```
Comment (ibc) [inline lib/RTCSession.js]: Honestly I do not like the idea of assuming that there is just a single audio track. Yup, I know that is common in VoIP and SIP regular calls, but forcing JsSIP to assume that by design is not something that makes me too happy.

@jmillan thoughts?

File: lib/RTCSession.js (hunk 1 above)
Comment (chukitow) [reply lib/RTCSession.js]: Good question, I can handle errors using offensive programing like this:
```
 const sender = this._getDTMFRTPSender();
sender && sender.insertDTMF(tone, duration, interToneGap);
```

File: lib/RTCSession.js (hunk 2 above)
Comment (jmillan) [reply lib/RTCSession.js]: I find it OK sending the DTMFs to the **first audio track found**.

Having said that, we no longer use `getLocalStreams()`. See [this].(https://github.com/versatica/JsSIP/blob/master/lib/RTCSession.js#L3346)

Also `createDTMFSender()` is obsoleted in favour of [RTCDTMFSender](https://www.w3.org/TR/webrtc/#dom-rtcdtmfsender)

**Comment by chukitow** [comment]
I updated the PR based on your comments @ibc @jmillan

File: lib/RTCSession.js [hunk 3]
```diff
@@ -3523,4 +3541,14 @@ module.exports = class RTCSession extends EventEmitter
...
   }
+
+  _getDTMFRTPSender()
+  {
+    const dtmfSender = this._connection.getSenders().find((sender) =>
+    {
+      return sender.track && sender.track.kind === 'audio';
+    });
+
+    return dtmfSender && dtmfSender.dtmf;
```
Comment (ibc) [inline lib/RTCSession.js]: Can we please avoid this `&&` usage and instead do:

```js
if (dtmfSender)
{
  return dtmfSender.dtmf;
}
```

?

Comment (chukitow) [reply lib/RTCSession.js]: Done @ibc

File: lib/RTCSession.js [hunk 4]
```diff
@@ -3523,4 +3541,17 @@ module.exports = class RTCSession extends EventEmitter
...
   }
+
+  _getDTMFRTPSender()
+  {
+    const dtmfSender = this._connection.getSenders().find((sender) =>
+    {
+      return sender.track && sender.track.kind === 'audio';
+    });
+
+    if (dtmfSender)
```
Comment (jmillan) [inline lib/RTCSession.js]: Is must also be checked whether the sender can send DTMFs via [canInsertDTMF](https://www.w3.org/TR/webrtc/#dom-rtcdtmfsender-caninsertdtmf) method

``` javascript
if (dtmfSender && dtmfSender.dtmf && dtmfSender.dtmf.canInsertDTMF)
{
 return dtmfSender.dtmf;
}

File: lib/RTCSession.js [hunk 5]
```diff
@@ -39,6 +39,8 @@ const C = {
  */
 const holdMediaTypes = [ 'audio', 'video' ];
 
+const RFC2833 = 'RFC2833';
```
Comment (ibc) [inline lib/RTCSession.js]: Do we really need this const? I think it provides no much value and we can get rid of it.

File: lib/RTCSession.js [hunk 6]
```diff
@@ -1042,13 +1045,28 @@ module.exports = class RTCSession extends EventEmitter
...
       else
       {
-        const dtmf = new RTCSession_DTMF(this);
+        switch (type)
+        {
+          case RFC2833: {
+            const sender = this._getDTMFRTPSender();
+
+            if (sender)
+            {
```
Comment (ibc) [inline lib/RTCSession.js]: Can we get rid of those unneeded brackets? BTW: have you tested `gulp lint`?

File: lib/RTCSession.js (hunk 5 above)
Comment (jmillan) [reply lib/RTCSession.js]: @ibc I agree with you. There is no need for this const.

To make it more clear to @chukitow,  do you want to share your suggestion for removing this constant?

**Comment by acharlop** [comment]
@ibc should probably close this one

**Comment by ibc** [comment]
Done in 3.4.5

**Pull request closed without merging**